### PR TITLE
BuildingTile version 1

### PR DESCRIPTION
**Pull request by Fremal31**
Version 1 BuildingTile.java class added with tests.

### CODE REVIEW

File: src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        BuildingTile buildingTile1 = new BuildingTile(buildingCards);
+    }
+
+    @Test
+    public void test_tryToPlaceFigures() {
+        buildingCards = setBuildingCards();
+        assertEquals(buildingCards.size(), 7);
+        BuildingTile buildingTile1 = new BuildingTile(buildingCards);
+        JSONObject state = new JSONObject(buildingTile1.state());
+        test_state(state, simpleBuilding, 7, emptyOrder);
```
Comment (Fremal31) [inline src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java:129]:
```suggestion
        for (int i = 7; i >= 1; i--) {
                assertEquals(buildingCards.size(), i);
                BuildingTile buildingTile1 = new BuildingTile(buildingCards);
                JSONObject state = new JSONObject(buildingTile1.state());
                test_state(state, simpleBuilding, i, emptyOrder);
                buildingCards.remove(0);
        }
```

File: src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        OptionalInt pointsFromBuilding = building.build(inputResources);
+        if (pointsFromBuilding.isPresent()) {
+            if (!player.playerBoard().takeResources(inputResources)) {
+                return ActionResult.FAILURE;
+            }
+
+            player.playerBoard().givePoints(pointsFromBuilding.getAsInt());
+            figures.remove(player.playerOrder());
+            buildingStack.pop();
```
[the remark this answers is not in the document]
Comment (Fremal31) [reply src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:67]: 👍

File: src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        assertEquals((building != null ? building.state() : "{}"), state.getString("currentBuilding"));
+        assertEquals(Integer.toString(expectedNum), state.getString("numberOfCardsInBuildingTile"));
+        assertEquals(playerOrder.toString(), state.getString("figures"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void test_TooFewCards() {
+        buildingCards = setBuildingCards();
+        buildingCards.remove(0);
+        assertEquals(buildingCards.size(), 6);
```
[the remark this answers is not in the document]
Comment (Fremal31) [reply src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java:102]: Upravil som aj ine testovacie metody.

File: src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        OptionalInt pointsFromBuilding = building.build(inputResources);
+        if (pointsFromBuilding.isPresent()) {
+            if (!player.playerBoard().takeResources(inputResources)) {
+                return ActionResult.FAILURE;
+            }
+
+            player.playerBoard().givePoints(pointsFromBuilding.getAsInt());
+            figures.remove(player.playerOrder());
+            player.playerBoard().giveEffect(Effect.BUILDING);
```
Comment (Fremal31) [inline src/main/java/sk/uniba/fmph/dcs/game_board/BuildingTile.java:67]:
```suggestion
            player.playerBoard().giveEffect(new ArrayList<Effect>(List.of(Effect.BUILDING));
```

File: src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+        BuildingTile buildingTile1 = new BuildingTile(buildingCards);
+    }
+
+    @Test
+    public void test_GoodInitialization() {
+        buildingCards = setBuildingCards();
+        assertEquals(buildingCards.size(), 7);
+        BuildingTile buildingTile1 = new BuildingTile(buildingCards);
+        JSONObject state = new JSONObject(buildingTile1.state());
+        test_state(state, simpleBuilding, 7, emptyOrder);
```
Comment (Fremal31) [inline src/test/java/sk/uniba/fmph/dcs/game_board/BuildingTileTest.java:110]:
```suggestion
        for (int i = 7; i >= 1; i--) {
    assertEquals(buildingCards.size(), i);
    BuildingTile buildingTile1 = new BuildingTile(buildingCards);
    JSONObject state = new JSONObject(buildingTile1.state());
    test_state(state, simpleBuilding, i, emptyOrder);
    buildingCards.remove(0);
}
```

**Review by rvadovic — APPROVED** [review]
ok

**Pull request merged**